test/integration/features/steps/parent_steps.py

```diff
@@ -0,0 +1,144 @@
+from behave import given, then, when
+from features.utilities.utils import (
+    get_lambda_info,
+    process_payload,
+    get_stored_events_from_dynamo_db,
+    search_dos_db,
+)
+from decimal import Decimal
+from features.utilities.change_events import get_change_event
+from features.utilities.log_stream import get_logs
+
+
+@given("a Changed Event is valid")
+def a_change_event_is_valid(context):
+    """Creates a valid change event"""
+    context.change_event = get_change_event()
+
+
+@given("a Changed Event has no matching DoS services")
+def a_change_event_with_no_matching_dos_services(context):
+    context.change_event = get_change_event()
+    context.change_event["ODSCode"] = "CODE1"
+
+
+@given("a Changed Event with invalid ODSCode is provided")
+def a_change_event_with_invalid_odscode(context):
+    """Creates a valid change event"""
+    context.change_event = get_change_event()
+    context.change_event["ODSCode"] = "FAKE6"
```

Comment:
How is this different to no matching DoS Service, why is CODE1 valid but not matching and FAKE6 is invalid

Reply:
Initial CODE1 step def function was to satisfy a scenario defined in the process flow doc of scenarios however this has now been redefined and a new step def created instead to validate no services match scenarios better.